### DISCUSSION TITLE
Querying `Select` choices?

### DISCUSSION THREAD

**Discussion by Wind-Explorer** [pocketbase/pocketbase#5540] [Q&A] [answered]
How can I retrieve the list of valid choices of a `Select` field in a collection?

For example for the field in the image below I would like to retrieve an array from it like

```
["web", "mobile", "desktop", ...]
```

<img width="557" alt="screenshot of a select field" src="https://github.com/user-attachments/assets/e320813b-38c8-4358-8c53-fcfcc57dded4">

As of now my implementation is an enum in my project that I have to manually update to keep up to date with the entries in my collections.

I have not been able to find relevant information in the documentation.

Any enlightment would be appreciated!

**Comment by ganigeorgiev** [comment, marked as answer]
There is no builtin method for this. You can either create a helper [View collection](https://pocketbase.io/docs/collections/#view-collection) and extract the field options from the `_collections` schema or create a custom route (_the latter is probably easier for this case_). Here is one such example with the JS `pb_hooks` (_I haven't tested it_):

```js
routerAdd("GET", "/select-options/:collection", (c) => {
    const collection =  $app.dao().findCollectionByNameOrId(c.pathParam("collection"))

    const result = {};

    for (let field of collection.schema.fields()) {
        if (field.type == "select") {
            result[field.name] = field.options.values || [];
        }
    }

    return c.json(200, result)
})
```

> my implementation is an enum in my project that I have to manually update to keep up to date with the entries in my collections.

If you want the select options to have labels different than the shown value you'll have to keep somewhere a static map with the labels anyway.

**Reply by Wind-Explorer** [reply]
> If you want the select options to have labels different than the shown value you'll have to keep somewhere a static map with the labels anyway.

Great point slipped past my mind for some reason 😂 guess that'd be the way I'm going with then.

Though thanks for the suggestion on View collection. I haven't looked into it yet I'll see what that's about.

**Reply by ganigeorgiev** [reply]
@aakashbashyal21 The initial code sample targets PocketBase <= v0.22.x. For v0.23+ the equivalent code sample could look something like (_I haven't tested it_):

```js
routerAdd("GET", "/select-options/{collection}", (e) => {
    const collection =  $app.findCollectionByNameOrId(e.request.pathValue("collection"))

    const result = {}

    for (let field of collection.fields) {
        if (field.type() == "select") {
            result[field.name] = field.values || [];
        }
    }

    return e.json(200, result)
})
```

_Edit: the user deleted their comment but the above reply remains valid._
